core/tests/test_utils.py

```diff
@@ -1125,6 +1129,75 @@ def save_new_default_exploration(
         exp_services.save_new_exploration(owner_id, exploration)
         return exploration
 
+    def set_interaction_for_state(self, state, interaction_id):
```

Comment:
Why are we doing all this in test_utils.py? Please try to get rid of this altogether -- we should be using the functions in the main code and not duplicate everything here.

Reply:
Is there a better place for this? This function needs to be used in 6 backend test files, so I though test_utils would be the proper place to put it (also, this function is used as a subprocess of another function already in test_utils)

Reply:
I'm fine with you having a new method here as long as it does not much more than calling a few functions in the main codebase. Does that clarify? What you have here currently is way too complicated to be a test utility function (which is typically just intended to consolidate a bunch of operations to make test setup simpler).